schema_based_on_diagram.sql

```diff
@@ -0,0 +1,49 @@
+CREATE TABLE patients (
+    id SERIAL NOT NULL PRIMARY KEY,
+    name VARCHAR(60) NOT NULL,
+    date_of_birth DATE NOT NULL
+);
+
+CREATE TABLE medical_histories (
+    id SERIAL NOT NULL PRIMARY KEY,
+    patient_id INTEGER NOT NULL REFERENCES patients(id),
+    admitted_at timestamp NOT NULL,
+    status VARCHAR(60) NOT NULL
+);
+
+CREATE TABLE invoices(
+    id SERIAL NOT NULL PRIMARY KEY,
+    total_amount DECIMAL(10,2) NOT NULL,
+    generated_at timestamp NOT NULL,
+    payed_at timestamp NOT NULL,
+    medical_history_id INTEGER NOT NULL REFERENCES medical_histories(id)
+);
+
+CREATE TABLE invoice_items (
+    id SERIAL NOT NULL PRIMARY KEY,
+    unit_price DECIMAL(10,2) NOT NULL,
+    quantity INTEGER NOT NULL,
+    total_price DECIMAL(10,2) NOT NULL,
+    invoice_id INTEGER REFERENCES invoices(id),
+    treatment_id INTEGER REFERENCES treatments(id)
```

Comment:
- [ ] [Optioanl] 👍 Good progress on your code! However, there's a small issue that I've noticed. When running your entire file, it throws an error because you're using the `id` of the `treatments` table before creating the `treatments` table itself. To resolve this issue, consider moving the creation of the `treatments` table before the line where you reference the `treatments` table's `id` in the foreign key constraint. This way, the table will exist when the foreign key is created. Keep up the good work! 🛠️👏

![image](https://github.com/ulises2607/Clinic-Database/assets/55840999/5599cc52-e373-4c2a-aa8c-8266a03b90ee)